### PR TITLE
gh-93124: Fix typos in comments

### LINKED ISSUE #93124: Typos found with codespell
Nothing major here, but I ran `codespell Lib/` and found a few typos. I will submit pull requests next.

```
                # here drained task ouside the barrier
                                    ^^^^^^

                # wit here to reach the `parties`
                  ^^^

        # signal.signal() can throw if the "main thread" doensn't have signals enabled
                                                         ^^^^^^^^

        # readline module is loaded, since test_builtin is not intented to test
                                                               ^^^^^^^^

        # When running under regtest, a seperate tempdir is used
                                        ^^^^^^^^

        invokations = []
            invokations.append(src)
        self.assertEqual(len(invokations), 9)

        trough that corresponds to VALUE or the current value if None is
        ^^^^^^

        # Parameterizes an already-parameterized object.
          ^^^^^^^^^^^^^

        """Parameterizes a generic class.
           ^^^^^^^^^^^^^
```

### PR DESCRIPTION
**Pull request by Akuli**
Fixes #93124

Automerge-Triggered-By: GH:rhettinger

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]

File: Lib/typing.py [hunk 1]
```diff
@@ -1359,7 +1359,7 @@ def __ror__(self, left):
 
     @_tp_cache
     def __getitem__(self, args):
-        # Parameterizes an already-parameterized object.
+        # Parametrizes an already-parameterized object.
```
Comment (JelleZijlstra) [inline Lib/typing.py]: I don't know which of "parameterize" and "parametrize" is correct, I think we may have discussed it in some earlier PR. But the two occurrences of the verb in this line should stay consistent. And if both spellings are acceptable, we should just stick with what we already have.

Comment (Akuli) [reply Lib/typing.py]: Both are apparently correct. I'll revert this part.

**Review by Fidget-Spinner — APPROVED** [review]

File: Lib/test/test_asyncio/test_locks.py [hunk 2]
```diff
@@ -1271,7 +1271,7 @@ async def coro():
                 # catch here waiting tasks
                 results1.append(True)
             else:
-                # here drained task ouside the barrier
+                # here drained task outside the barrier
```
Comment (JelleZijlstra) [inline Lib/test/test_asyncio/test_locks.py:1274]: Not entirely sure what this comment is trying to say

**Review by JelleZijlstra — APPROVED** [review]

**Review by rhettinger — APPROVED** [review]

**Comment by rhettinger** [comment]
Closing and reopening so that the bots will rerun.

**Comment by Akuli** [comment]
No idea. The bot just posted lots of comments when I pushed a new commit :D

**Pull request merged**